### DISCUSSION TITLE
Comparing Messages with `Eq`?

### DISCUSSION THREAD

**Discussion by hpux735** [diwic/dbus-rs#330] [Q&A]
Is there a way to compare instances of the `Message` type?  It's hard to write unit tests to make sure that implementor's code is working as expected without being able to write things like `assert_eq!(sample_message, test_message`, for example.

**Comment by diwic** [comment]
Good idea. No, I don't think so at the moment. I guess you could debug print both as a workaround and compare the debug prints. 
Patches welcome :-)

**Reply by diwic** [reply]
Although at second thought, that requires some careful thinking. E g, a message that you received over the bus would have the "sender" and "id" fields filled in. A message you constructed yourself would not have that. That seems problematic for asserting that two messages are equal.

**Reply by hpux735** [reply]
Yah, that makes sense.  I tried with PropMap too (which I think is the better option), but I'm just way to new (2 weeks, lol) at Rust to get my head around comparing the `dyn RefArg` stuff.
